### DISCUSSION TITLE
Conditional ToggleColumn

### DISCUSSION THREAD

**Discussion by clinvest** [filamentphp/filament#11139] [Help] [answered]
### Package

Table builder

### Package Version

v3.0.0

### How can we help you?

Hi all,

My $record is always null in the function below. Shouldn't it be working? Am I doing something wrong? I correctly see all my records in my table.

`ToggleColumn::make('is_artist_paid')->visible(function (?Model $record) { return $record?->management_status->value === 'artist_payment_pending'; }),`

Thank you in advance,

**Comment by leandrocfe** [comment, marked as answer]
You can't show/hide columns by row. Try to use `->disabled()`

**Comment by clinvest** [comment]
Thank you Leandro!
